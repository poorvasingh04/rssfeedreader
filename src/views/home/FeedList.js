import React from 'react';
import {
  FlatList,
  TouchableHighlight,
} from 'react-native';

function FeedList({ navigation }) {
  var feedsData = [];

  openFeed = (feed) => {

  }

  renderItem = ({ item }) => {
    return (
      <TouchableHighlight
        onPress={() => openFeed(item)}
      >
        <Text>Hello</Text>
      </TouchableHighlight>
    );
  }

  return (
    <FlatList
      keyExtractor={(_, index) => index.toString()}
      renderItem={renderItem}
      data={feedsData}
    />

  );
}

export default FeedList;