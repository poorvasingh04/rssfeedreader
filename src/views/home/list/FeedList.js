import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import {
  FlatList,
} from 'react-native';
import { getFeedList } from '../../../services/actions';
import Feed from '../feed/Feed';
import SCREENS from '../../../constants/SCREENS';

function FeedList({ navigation }) {
  const dispatch = useDispatch();

  const feedsData = useSelector(state => state.FeedList.data);

  useEffect(() => {
    let cancel = false;
    const runEffect = async () => {      
      if (cancel) {
        return;
      }
     
      dispatch(getFeedList());
    };
    runEffect();
  
    return () => {
      cancel = true;
    }
  }, []);

  openFeed = (feed) => {
    if(navigation) navigation.navigate(
      SCREENS.FEED_DETAILS,
      {
        feed,
      }
    );
  }

  renderItem = ({ item }) => <Feed item={item} openFeed={openFeed} />

  return (
    <FlatList
      keyExtractor={(item) => item.id}
      renderItem={renderItem}
      data={feedsData}
    />

  );
}

export default FeedList;