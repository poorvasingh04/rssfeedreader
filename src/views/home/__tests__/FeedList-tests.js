import 'react-native';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import FeedList from '../list/FeedList';
import { FlatList, Text, TouchableOpacity } from 'react-native';

jest.mock('react-native-webview', () => 'WebView');

describe('FeedList test cases-----', () => {
  const mockStore = configureStore()
  let store, wrapper;
  const navigateAction = jest.fn();
  const navigation = {
    navigate: (props, data) => navigateAction(props, data),
  }

  const firstFeed = {
    id: '1', name: 'Google',
  };

  beforeEach(() => {
    store = mockStore({
      FeedList: {
        data: [firstFeed],
      }
    });
    wrapper = renderer.create(
      <Provider store={store}>
        <FeedList navigation={navigation} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders correct number of components', () => {
    const testInstance = wrapper.root;
    expect(testInstance.findByType(FlatList).props.data.length).toBe(1);
  });

  it('test render item', () => {
    const testInstance = wrapper.root;
    expect(testInstance.findAllByType(TouchableOpacity).length).toBe(1);
    expect(testInstance.findAllByType(Text).length).toBe(1);
  });

  it('should open feed on feed tap', () => {
    const feedCard = wrapper.root.findAllByType(TouchableOpacity)[0];
    feedCard.props.onPress();
    expect(navigateAction).toBeCalledWith('Feed Details', { feed: firstFeed });
  });
  
})