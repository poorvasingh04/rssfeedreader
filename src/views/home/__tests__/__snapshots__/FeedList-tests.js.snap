// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedList test cases----- renders correctly 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "id": "1",
        "name": "Google",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 10,
            "elevation": 4,
            "margin": 10,
            "opacity": 1,
            "padding": 20,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#0645AD",
              "fontSize": 16,
              "textDecorationLine": "underline",
            }
          }
        >
          Google
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
